src/sqlfluff/rules/L025.py

```diff
@@ -78,8 +81,16 @@ def _eval(self, context: RuleContext) -> EvalResultType:
             alias: AliasInfo
             for alias in query.aliases:
                 if alias.aliased and alias.ref_str not in query.tbl_refs:
-                    # Unused alias. Report and fix.
-                    violations.append(self._report_unused_alias(alias))
+                    # Don't warn or remove aliases for table functions, e.g.
+                    # Snowflake FLATTEN(). They are special because unlike real
+                    # "tables", they don't have a natural name to fall back to
+                    # if the alias is removed. Also, removing them can in turn
+                    # trigger false positives with L028.
+                    if not alias.table_expression or not has_table_function(
+                        alias.table_expression, context.dialect
+                    ):
+                        # Unused alias. Report and fix.
+                        violations.append(self._report_unused_alias(alias))
```

Comment:
Important decision here -- we're not going to remove unused aliases for table functions, even if they're unused. I'm not sure this is the perfect solution, but it seems good enough for now.